drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/DriftNettyMethodInvokerFactory.java

```diff
@@ -92,38 +98,50 @@ public DriftNettyMethodInvokerFactory(
         this.defaultSocksProxy = Optional.ofNullable(factoryConfig.getSocksProxy());
 
         connectionPoolMaintenanceExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("drift-connection-maintenance"));
-
-        ConnectionManager connectionManager = new ConnectionFactory(group, sslContextFactory, allocator, factoryConfig);
-        if (factoryConfig.isConnectionPoolEnabled()) {
-            connectionManager = new ConnectionPool(
-                    connectionManager,
-                    group,
-                    factoryConfig.getConnectionPoolMaxSize(),
-                    factoryConfig.getConnectionPoolMaxConnectionsPerDestination(),
-                    factoryConfig.getConnectionPoolIdleTimeout(),
-                    connectionPoolMaintenanceExecutor);
-        }
-        this.connectionManager = connectionManager;
+        connectionFactory = new ConnectionFactory(group, sslContextFactory, allocator, factoryConfig);
     }
 
     @Override
     public MethodInvoker createMethodInvoker(I clientIdentity)
     {
-        ConnectionParameters clientConfig = toConnectionConfig(clientConfigurationProvider.apply(clientIdentity));
+        DriftNettyClientConfig driftNettyClientConfig = clientConfigurationProvider.apply(clientIdentity);
+        ConnectionParameters clientConfig = toConnectionConfig(driftNettyClientConfig);
 
         // validate ssl context configuration is valid
         clientConfig.getSslContextParameters()
                 .ifPresent(sslContextParameters -> sslContextFactory.get(sslContextParameters).get());
 
+        ConnectionManager connectionManager = getConnectionManager(clientIdentity, driftNettyClientConfig);
         return new DriftNettyMethodInvoker(clientConfig, connectionManager, group);
     }
 
+    public ConnectionManager getConnectionManager(I clientIdentity, DriftNettyClientConfig driftNettyClientConfig)
+    {
+        boolean connectionPoolEnabled = firstNonNull(driftNettyClientConfig.getConnectionPoolEnabled(), factoryConfig.isConnectionPoolEnabled());
```

Comment:
Should both be set as true to enable connection pool?  Otherwise, we don't have global control.

Reply:
Basically the "per client" config takes the precedence. If "per client" config is not set - the global configuration will be respected